pkg/client/didexchange/client.go

```diff
@@ -272,7 +272,15 @@ func (c *Client) GetConnectionAtState(connectionID, stateID string) (*Connection
 }
 
 // RemoveConnection removes connection record for given id
-func (c *Client) RemoveConnection(id string) error {
-	// TODO https://github.com/hyperledger/aries-framework-go/issues/553 RemoveConnection from did exchange service
+func (c *Client) RemoveConnection(connectionID string) error {
+	err := c.connectionStore.RemoveConnection(connectionID)
+	if errors.Is(err, storage.ErrDataNotFound) {
+		return err
+	}
+
+	if err != nil {
+		return fmt.Errorf("cannot remove connection from the store: connectionid=%s err=%s", connectionID, err)
+	}
+
```

Comment:
No need to return two different error messages. 
Use `%w` to wrap an error. It allows you to get an initial error by calling errors.Unwrap().

Reply:
Got it and fixed now.